homeassistant/setup.py

```diff
@@ -289,14 +290,34 @@ def log_error(msg: str, exc_info: Exception | None = None) -> None:
         log_error(str(err))
         return False
 
+    if debug := _LOGGER.isEnabledFor(logging.DEBUG):
+        start = timer()
+
+    load_executor = (
+        integration.import_executor
+        and f"hass.components.{domain}" not in sys.modules
+        and f"custom_components.{domain}" not in sys.modules
+    )
     # Some integrations fail on import because they call functions incorrectly.
     # So we do it before validating config to catch these errors.
+
     try:
-        component = integration.get_component()
+        if load_executor:
```

Comment:
Is it weird that we have this logic here and not inside `integration.get_component` ?

Reply:
I think it would be too big of a change to make `get_component` a coro

Reply:
I guess its not so bad. I could break it up like get_platform

Reply:
I do like that better f1d8860846851a067c4b9c0aa48796b63071988e